lib/apiservers/portlayer/restapi/handlers/containers_handlers.go

```diff
@@ -71,7 +71,11 @@ func (handler *ContainersHandlersImpl) CreateHandler(params containers.CreatePar
 	log.Debugf("Env: %#v", params.CreateConfig.Env)
 	log.Debugf("WorkingDir: %#v", params.CreateConfig.WorkingDir)
 
-	id := exec.GenerateID().String()
+	containerID := exec.GenerateID()
+	id := containerID.String()
+
+	// inject hostname into environment
+	params.CreateConfig.Env = append(params.CreateConfig.Env, fmt.Sprintf("HOSTNAME=%s", containerID.TruncateID()))
```

Comment:
Agreed that it's a personality thing, but we don't get the ID back until the container is already created (and `ENV` already injected). I tossed around the idea of creating the ID in the personality itself, but @hmahmood objected on the grounds that it should be generated in the portlayer.

This is why I originally didn't have it in the environment vars, but only in the local shell vars. :)